client/lib/purchases/index.js

```diff
@@ -173,6 +179,7 @@ function isRemovable( purchase ) {
 	}
 
 	return (
+		isJetpackPlan( purchase ) ||
```

Comment:
This is the alternative approach to #25586, which should have much more limited impact.

cc: @aidvu @klimeryk

Reply:
Right! The other conditions seem oriented at different kinds of purchases. I can't imagine a reason why a user who wants to remove the purchase of a Jetpack plan should be prevented from doing so.